sklearn/utils/stats.py

```diff
@@ -66,4 +68,15 @@ def _weighted_percentile(array, sample_weight, percentile=50):
     col_index = np.arange(array.shape[1])
     percentile_in_sorted = sorted_idx[percentile_idx, col_index]
     percentile = array[percentile_in_sorted, col_index]
+
+    # percentiles that point to nan values are redirected to the next lower value:
+    while bool(np.isnan(percentile).any()) and (
+        percentile_idx[np.isnan(percentile)].all() > 0
```

Comment:
Should `bool(np.isnan(percentile).any())` be an in condition withing the loop for better readability? It actually doesn't construct the looping condition.

Reply:
I don't understand the `percentile_idx[np.isnan(percentile)].all() > 0` condition. `percentile_idx[np.isnan(percentile)]` is an array of (non-binary) integers. So calling `.all()` on it should be equivalent to `0 not in percentile_idx[np.isnan(percentile)]` but then I do not see the need to compare that boolean to 0 with `> 0`.

Maybe you meant

```python
(percentile_idx[np.isnan(percentile)] > 0).all()
```

instead?